test/integration/create.reservation.test.js

```diff
@@ -0,0 +1,108 @@
+process.env.NODE_ENV = 'test';
+const request = require('supertest');
+const expect = require('chai').expect;
+const knex = require('../../db');
+const moment = require('moment');
+const sinon = require('sinon');
+const ReservationModel = require('../../app/models/Reservations');
+
+describe('Create reservation', () => {
+    let server;
+    let reservation;
+    let stub;
+
+    const exec = () => {
+        return request(server).post('/api/reservations');
+    };
+
+    beforeEach(async () => {
+        server = require('../../index');
+        await knex.migrate.rollback();
+        await knex.migrate.latest();
+        await knex.seed.run();
+        reservation = new ReservationModel();
+    });
+
+    afterEach(async () => {
+        await knex.migrate.rollback();
+        await server.close();
+        reservation = null;
+        stub = sinon.reset();
+    });
+
+    after(() => {
+        stub = sinon.restore();
```

Comment:
`reset` and `restore` should call on stub
https://sinonjs.org/releases/v6.3.4/stubs/